schematic/schemas/data_model_json_schema.py

```diff
@@ -0,0 +1,357 @@
+							)
+
+						json_schema["properties"].update(schema_valid_vals)
+						# add node to required fields
+						json_schema["required"] += [node_display_name]
+
+					elif process_node in root_dependencies:
+						# node doesn't have conditionals and is not required; it belongs in the schema only if it is in root's dependencies
+
+						if node_range:
+							schema_valid_vals = self.get_range_schema(
+								node_range=node_range_d, node_name=node_display_name, blank=True
+							)
+
+							if node_validation_rules:
+								if rule_in_rule_list("list", node_validation_rules):
+									schema_valid_vals = self.get_array_schema(
+										node_range=node_range_d, node_name=node_display_name, blank=True
+									)
+
+						else:
+							schema_valid_vals = {node_display_name: {}}
+
+						json_schema["properties"].update(schema_valid_vals)
+
+					else:
+						# node doesn't have conditionals and it is not required and it is not a root dependency
+						# the node doesn't belong in the schema
+						# do not add to processed nodes since its conditional may be traversed at a later iteration (though unlikely for most schemas we consider)
+						node_is_processed = False
+
+				# add process node as a conditional to its dependencies
+				node_dependencies_d = self.DME.get_nodes_display_names(
+					node_list=node_dependencies
+				)
+
+				for dep in node_dependencies_d:
+					if not dep in reverse_dependencies:
+						reverse_dependencies[dep] = []
+
+					reverse_dependencies[dep].append(node_display_name)
+
+				# add nodes found as dependencies and range of this processed node
+				# to the list of nodes to be processed
+				nodes_to_process += node_range
+				nodes_to_process += node_dependencies
+
+				# if the node is processed add it to the processed nodes set
+				if node_is_processed:
+					processed_nodes.append(process_node)
+
+			# if the list of nodes to process is not empty
+			# set the process node the next remaining node to process
+			if nodes_to_process:
+				process_node = nodes_to_process.pop(0)
+			else:
+				# no more nodes to process
+				# exit the loop
+				break
+
+		logger.info("JSON schema successfully generated from schema.org schema!")
+
+		# if no conditional dependencies were added we can't have an empty 'AllOf' block in the schema, so remove it
+		if not json_schema["allOf"]:
+			del json_schema["allOf"]
+
+		# If no config value and SchemaGenerator was initialized with
+		# a JSON-LD path, construct
+		if self.jsonld_path is not None:
+			self.jsonld_path_root, jsonld_ext = os.path.splitext(self.jsonld_path)
+			prefix = self.jsonld_path_root
+			prefix_root, prefix_ext = os.path.splitext(prefix)
+			if prefix_ext == ".model":
+				prefix = prefix_root
+			json_schema_log_file = f"{prefix}.{source_node}.schema.json"
+
+		logger.info(
+			"The JSON schema file can be inspected by setting the following "
+			"nested key in the configuration: (model > input > log_location)."
```

Comment:
This should be "model > location" because we updated the config

Reply:
@linglp updated, but will actually will comment out the logging for now, since the change to the config i think actually broke the saving of the json schema file. There is already a ticket to address this in the backlog.

Reply:
The logging can be un-commented when/if its fixed.